adapters/httpapi_winhttp.c

```diff
@@ -476,7 +479,20 @@ static HTTPAPI_RESULT InitiateWinhttpRequest(HTTP_HANDLE_DATA* handleData, HTTPA
     ))
     {
         result = HTTPAPI_SET_X509_FAILURE;
-        LogErrorWinHTTPWithGetLastErrorAsString("unable to WinHttpSetOption");
+        LogErrorWinHTTPWithGetLastErrorAsString("unable to WinHttpSetOption (WINHTTP_OPTION_CLIENT_CERT_CONTEXT)");
+        (void)WinHttpCloseHandle(*requestHandle);
+        *requestHandle = NULL;
+    }
+    else if ((handleData->x509SchannelHandle != NULL) &&
```

Comment:
why are we checking `handleData->x509SchannelHandle` here?  Shouldn't this be set unconditionally?